### PR TITLE
58714_Static Pages

### PR DESCRIPTION
**Pull request by ahammad-reyad**
Added static pages

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,47 +1,30 @@
...
-
-# Use ActiveModel has_secure_password
-# gem 'bcrypt', '~> 3.1.7'
-
-# Use Unicorn as the app server
-# gem 'unicorn'
-
-# Use Capistrano for deployment
-# gem 'capistrano-rails', group: :development
+gem 'rails',        '4.2.2'
```
Comment (dieunb) [inline Gemfile]: pls use double quote instead of single quote. Use latest rails version. (4.2.6)

File: Gemfile [hunk 2]
```diff
@@ -1,47 +1,30 @@
...
+gem 'bootstrap-sass', '3.2.0.0'
```
Comment (dieunb) [inline Gemfile]: pls use double quote instead of single quote

File: Gemfile [hunk 3]
```diff
@@ -1,47 +1,30 @@
...
+gem 'rails-i18n', '~> 4.0.0'
```
Comment (dieunb) [inline Gemfile]: pls use double quote instead of single quote

File: Gemfile [hunk 4]
```diff
@@ -1,47 +1,30 @@
...
+gem 'sass-rails',   '5.0.2'
```
Comment (dieunb) [inline Gemfile]: pls use double quote instead of single quote

File: Gemfile [hunk 5]
```diff
@@ -1,47 +1,30 @@
...
+gem 'uglifier',     '2.5.3'
```
Comment (dieunb) [inline Gemfile]: pls use double quote instead of single quote

File: Gemfile [hunk 6]
```diff
@@ -1,47 +1,30 @@
...
+gem 'coffee-rails', '4.1.0'
```
Comment (dieunb) [inline Gemfile]: pls use double quote instead of single quote

File: app/assets/javascripts/static_pages.coffee [hunk 7]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (dieunb) [inline app/assets/javascripts/static_pages.coffee]: remove this file. pls use `git rm -f app/assets/javascripts/static_pages.coffee`

File: app/assets/stylesheets/static_pages.css [hunk 8]
```diff
@@ -0,0 +1,4 @@
+/*
```
Comment (dieunb) [inline app/assets/stylesheets/static_pages.css]: remove this file `git rm -f app/assets/stylesheets/static_pages.css
`

File: app/helpers/application_helper.rb [hunk 9]
```diff
@@ -1,2 +1,12 @@
 module ApplicationHelper
+
+  # Returns the full title on a per-page basis.
```
Comment (dieunb) [inline app/helpers/application_helper.rb]: remove all remark

File: app/helpers/application_helper.rb [hunk 10]
```diff
@@ -1,2 +1,12 @@
...
+  def full_title(page_title = '')
```
Comment (dieunb) [inline app/helpers/application_helper.rb]: remove `()`

File: app/helpers/static_pages_helper.rb [hunk 11]
```diff
@@ -0,0 +1,2 @@
+module StaticPagesHelper
```
Comment (dieunb) [inline app/helpers/static_pages_helper.rb]: remove this file `git rm -f app/helpers/static_pages_helper.rb`

File: app/views/layouts/application.html.erb [hunk 12]
```diff
@@ -1,14 +1,18 @@
...
-  <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
-  <%= csrf_meta_tags %>
-</head>
-<body>
-
-<%= yield %>
-
-</body>
+  <head>
+    <title><%= full_title(yield(:title)) %></title>
```
Comment (dieunb) [inline app/views/layouts/application.html.erb]: remove `()`

File: app/views/layouts/application.html.erb [hunk 13]
```diff
@@ -1,14 +1,18 @@
...
+    <%= stylesheet_link_tag "application", media: "all",
+                                           "data-turbolinks-track" => true %>
```
Comment (dieunb) [inline app/views/layouts/application.html.erb]: in ruby 2.2.0, you can use `"data-turbolinks-track": true `

File: app/views/layouts/application.html.erb [hunk 14]
```diff
@@ -1,14 +1,18 @@
...
+    <%= javascript_include_tag "application", "data-turbolinks-track" => true %>
```
Comment (dieunb) [inline app/views/layouts/application.html.erb]: in ruby 2.2.0, you can use `"data-turbolinks-track": true `

File: app/views/layouts/application.html.erb [hunk 15]
```diff
@@ -1,14 +1,18 @@
...
+    <%= csrf_meta_tags %>
+    <%= render 'layouts/shim' %>
```
Comment (dieunb) [inline app/views/layouts/application.html.erb]: double quote

File: app/views/layouts/application.html.erb [hunk 16]
```diff
@@ -1,14 +1,18 @@
...
+  </head>
+  <body>
+    <%= render 'layouts/header' %>
```
Comment (dieunb) [inline app/views/layouts/application.html.erb]: double quote

File: app/views/layouts/application.html.erb [hunk 17]
```diff
@@ -1,14 +1,18 @@
...
+    <div class="container">
+      <%= yield %>
+      <%= render 'layouts/footer' %>
```
Comment (dieunb) [inline app/views/layouts/application.html.erb]: double quote

File: app/views/static_pages/about.html.erb [hunk 18]
```diff
@@ -0,0 +1,5 @@
+<% provide(:title, t(:about)) %>
```
Comment (dieunb) [inline app/views/static_pages/about.html.erb]: remove `()`

File: app/views/static_pages/about.html.erb [hunk 19]
```diff
@@ -0,0 +1,5 @@
...
+<h1><%= t(:app_name) %></h1>
```
Comment (dieunb) [inline app/views/static_pages/about.html.erb]: remove `()`

File: app/views/static_pages/help.html.erb [hunk 20]
```diff
@@ -0,0 +1,5 @@
+<% provide(:title, t(:help)) %>
```
Comment (dieunb) [inline app/views/static_pages/help.html.erb]: remove `()`

File: app/views/static_pages/help.html.erb [hunk 21]
```diff
@@ -0,0 +1,5 @@
...
+<h1><%= t(:app_name) %></h1>
```
Comment (dieunb) [inline app/views/static_pages/help.html.erb]: remove `()`

File: app/views/static_pages/home.html.erb [hunk 22]
```diff
@@ -0,0 +1,4 @@
+<div class="center jumbotron">
+  <h1><%= t(:welcome_msg) %></h1>
```
Comment (dieunb) [inline app/views/static_pages/home.html.erb]: remove `()`

File: app/views/static_pages/home.html.erb [hunk 23]
```diff
@@ -0,0 +1,4 @@
...
+  <%= link_to "Sign up now!", '#', class: "btn btn-lg btn-primary" %>
```
Comment (dieunb) [inline app/views/static_pages/home.html.erb]: i18n

File: config/routes.rb [hunk 24]
```diff
@@ -1,56 +1,5 @@
...
-  #   resources :posts, concerns: :toggleable
-  #   resources :photos, concerns: :toggleable
-
-  # Example resource route within a namespace:
-  #   namespace :admin do
-  #     # Directs /admin/products/* to Admin::ProductsController
-  #     # (app/controllers/admin/products_controller.rb)
-  #     resources :products
-  #   end
+  root             'static_pages#home'
```
Comment (dieunb) [inline config/routes.rb]: double quote

File: config/routes.rb [hunk 25]
```diff
@@ -1,56 +1,5 @@
...
+  get 'help'    => 'static_pages#help'
```
Comment (dieunb) [inline config/routes.rb]: double quote

File: config/routes.rb [hunk 26]
```diff
@@ -1,56 +1,5 @@
...
+  get 'about'   => 'static_pages#about'
```
Comment (dieunb) [inline config/routes.rb]: double quote

File: app/views/layouts/_footer.html.erb [hunk 27]
```diff
@@ -0,0 +1,12 @@
+<footer class = "footer" >
+  <small>
+    <%= t :app_name %>
+    <%= t :author_name %>
+  </small>
+  <nav>
+    <ul>
+      <li><%= link_to "About",   about_path %></li>
```
Comment (dieunb) [inline app/views/layouts/_footer.html.erb]: i18n

File: app/views/layouts/_footer.html.erb [hunk 28]
```diff
@@ -0,0 +1,12 @@
...
+      <li><%= link_to "Contact", "#" %></li>
```
Comment (dieunb) [inline app/views/layouts/_footer.html.erb]: i18n

File: app/views/layouts/_header.html.erb [hunk 29]
```diff
@@ -0,0 +1,12 @@
+<header class = "navbar navbar-fixed-top navbar-inverse" >
+  <div class = "container" >
+    <%= link_to "sample app", "#", id: "logo" %>
```
Comment (MuhammadTamzid) [inline app/views/layouts/_header.html.erb]: i18n

File: app/views/layouts/application.html.erb [hunk 30]
```diff
@@ -1,14 +1,18 @@
...
-</head>
-<body>
-
-<%= yield %>
-
-</body>
+  <head>
+    <title><%= full_title yield :title %></title>
+    <%= stylesheet_link_tag "application", media: "all",
+                                           "data-turbolinks-track": true %>
```
Comment (MuhammadTamzid) [inline app/views/layouts/application.html.erb]: indent

File: app/helpers/application_helper.rb [hunk 31]
```diff
@@ -1,2 +1,11 @@
 module ApplicationHelper
+
+  def full_title page_title = ""
+    base_title = t(:app_name)
```
Comment (MuhammadTamzid) [inline app/helpers/application_helper.rb]: t "app_name"

File: app/views/layouts/_footer.html.erb [hunk 32]
```diff
@@ -0,0 +1,12 @@
+<footer class = "footer" >
+  <small>
+    <%= t :app_name %>
+    <%= t :author_name %>
+  </small>
+  <nav>
+    <ul>
+      <li><%= link_to t(:about),   about_path %></li>
```
Comment (MuhammadTamzid) [inline app/views/layouts/_footer.html.erb]: remove space

File: app/views/layouts/_header.html.erb [hunk 33]
```diff
@@ -0,0 +1,12 @@
+<header class = "navbar navbar-fixed-top navbar-inverse" >
+  <div class = "container" >
+    <%= link_to link_to t(:app_name), "#", id: "logo" %>
+    <nav>
+      <ul class="nav navbar-nav navbar-right">
+        <li><%= link_to t(:home),    root_path %></li>
```
Comment (MuhammadTamzid) [inline app/views/layouts/_header.html.erb]: remove space

File: app/views/layouts/_header.html.erb [hunk 34]
```diff
@@ -0,0 +1,12 @@
...
+        <li><%= link_to t(:help),    help_path %></li>
```
Comment (MuhammadTamzid) [inline app/views/layouts/_header.html.erb]: remove space

File: app/views/layouts/application.html.erb [hunk 35]
```diff
@@ -1,14 +1,18 @@
...
-</head>
-<body>
-
-<%= yield %>
-
-</body>
+  <head>
+    <title><%= full_title yield :title %></title>
+    <%= stylesheet_link_tag "application", media: "all",
+    "data-turbolinks-track": true %>
```
Comment (MuhammadTamzid) [inline app/views/layouts/application.html.erb]: two space

File: app/views/layouts/application.html.erb [hunk 36]
```diff
@@ -1,14 +1,19 @@
...
-</head>
-<body>
-
-<%= yield %>
-
-</body>
+  <head>
+    <title><%= full_title yield :title %></title>
+    <%= stylesheet_link_tag "application", media: "all",
+        "data-turbolinks-track": true %>
```
Comment (MuhammadTamzid) [inline app/views/layouts/application.html.erb]: indent two space

File: app/views/layouts/application.html.erb [hunk 37]
```diff
@@ -1,14 +1,19 @@
...
+    <%= javascript_include_tag "application",
+        "data-turbolinks-track": true %>
```
Comment (MuhammadTamzid) [inline app/views/layouts/application.html.erb]: indent two space

File: app/views/layouts/_footer.html.erb [hunk 38]
```diff
@@ -0,0 +1,12 @@
+<footer class = "footer" >
```
Comment (dieunb) [inline app/views/layouts/_footer.html.erb]: for html 
```
class="footer">
```

File: app/views/layouts/_header.html.erb [hunk 39]
```diff
@@ -0,0 +1,12 @@
+<header class = "navbar navbar-fixed-top navbar-inverse" >
```
Comment (dieunb) [inline app/views/layouts/_header.html.erb]:
```
<header class="navbar navbar-fixed-top navbar-inverse">
```

pls, update this for all your code

File: app/views/layouts/_header.html.erb [hunk 40]
```diff
@@ -0,0 +1,12 @@
...
+  <div class = "container" >
```
Comment (MuhammadTamzid) [inline app/views/layouts/_header.html.erb]: ``
<div class="container">
``

File: app/views/static_pages/home.html.erb [hunk 41]
```diff
@@ -0,0 +1,4 @@
+<div class = "center jumbotron" >
```
Comment (MuhammadTamzid) [inline app/views/static_pages/home.html.erb]: ``
<div class="center jumbotron">
``

File: app/views/layouts/application.html.erb [hunk 42]
```diff
@@ -1,14 +1,19 @@
...
+    <%= stylesheet_link_tag "application", media: "all",
+      "data-turbolinks-track": true %>
+    <%= javascript_include_tag "application",
+      "data-turbolinks-track": true %>
+    <%= csrf_meta_tags %>
+    <%= render "layouts/shim" %>
+  </head>
+  <body>
+    <%= render "layouts/header" %>
+    <div class = "container">
```
Comment (MuhammadTamzid) [inline app/views/layouts/application.html.erb]: ``
<div class="container">
``

**Pull request merged**